src/main/io/gps_ublox.c

```diff
@@ -532,81 +595,80 @@ static bool gpsNewFrameUBLOX(uint8_t data)
     return parsed;
 }
 
-// Send UBLOX binary command data and wait until it is completely transmitted
-static bool ubxTransmitAutoConfigCommands(const uint8_t * ubxCmdBuf, uint8_t ubxCmdSize) {
-    while (serialTxBytesFree(gpsState.gpsPort) > 0) {
-        if (gpsState.autoConfigPosition < ubxCmdSize) {
-            serialWrite(gpsState.gpsPort, ubxCmdBuf[gpsState.autoConfigPosition]);
-            gpsState.autoConfigPosition++;
-        }
-        else if (isSerialTransmitBufferEmpty(gpsState.gpsPort)) {
-            gpsState.autoConfigStep++;
-            gpsState.autoConfigPosition = 0;
-            return true;
-        }
-        else {
-            return false;
-        }
-    }
-
-    return false;
-}
-
 static bool gpsConfigure(void)
 {
     switch (gpsState.autoConfigStep) {
     case 0: // NAV5
         switch (gpsState.gpsConfig->dynModel) {
             case GPS_DYNMODEL_PEDESTRIAN:
-                ubxTransmitAutoConfigCommands(ubloxInit_NAV5_Pedestrian, sizeof(ubloxInit_NAV5_Pedestrian));
+                configureNAV5(UBX_DYNMODEL_PEDESTRIAN, UBX_FIXMODE_AUTO);
                 break;
             case GPS_DYNMODEL_AIR_1G:   // Default to this
             default:
-                ubxTransmitAutoConfigCommands(ubloxInit_NAV5_Airborne1G, sizeof(ubloxInit_NAV5_Airborne1G));
+                configureNAV5(UBX_DYNMODEL_AIR_1G, UBX_FIXMODE_AUTO);
                 break;
             case GPS_DYNMODEL_AIR_4G:
-                ubxTransmitAutoConfigCommands(ubloxInit_NAV5_Airborne4G, sizeof(ubloxInit_NAV5_Airborne4G));
+                configureNAV5(UBX_DYNMODEL_AIR_4G, UBX_FIXMODE_AUTO);
                 break;
         }
+        gpsState.autoConfigStep++;
         break;
 
     case 1: // NAVX5 - skip
-        //ubxTransmitAutoConfigCommands(ubloxInit_NAVX5, sizeof(ubloxInit_NAVX5));
         gpsState.autoConfigStep++;
         break;
 
     case 2: // Disable NMEA messages
-        ubxTransmitAutoConfigCommands(ubloxInit_MSG_NMEA, sizeof(ubloxInit_MSG_NMEA));
+        configureMSG(MSG_CLASS_NMEA, MSG_NMEA_GGA, 0);
```

Comment:
Would it be better to split this bunch of messages over different `autoConfigStep` values?